portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/ExportResult.java

```diff
@@ -1,23 +1,32 @@
 package org.dataportabilityproject.spi.transfer.provider;
 
+import com.google.common.base.Preconditions;
 import org.dataportabilityproject.spi.transfer.types.ContinuationData;
 import org.dataportabilityproject.types.transfer.models.DataModel;
 
-/** The result of an item export operation. */
+/**
+ * The result of an item export operation.
+ */
 public class ExportResult<T extends DataModel> {
+
   public static final ExportResult CONTINUE = new ExportResult(ResultType.CONTINUE);
   public static final ExportResult END = new ExportResult(ResultType.CONTINUE);
+  private static final String MUST_HAVE_THROWABLE = "ExportResult with ResultType = ERROR must hold a throwable";
+
   private ResultType type;
   private String message;
   private T exportedData;
   private ContinuationData continuationData;
+  private Throwable throwable;
```

Comment:
nit: can you document that this should be null if nothing was thrown during the export?